client/asset/btc/rpcclient.go

```diff
@@ -789,6 +790,35 @@ func (wc *rpcClient) sendToAddress(address string, value, feeRate uint64, subtra
 	return chainhash.NewHashFromStr(txid)
 }
 
+func (wc *rpcClient) estimateSendTxFee(tx *wire.MsgTx, feeRate uint64, subtract bool) (txfee uint64, err error) {
+	txBytes, err := wc.serializeTx(tx)
+	if err != nil {
+		return 0, fmt.Errorf("tx serialization error: %w", err)
+	}
+
+	// 1e-5 = 1e-8 for satoshis * 1000 for kB.
+	feeRateOption := float64(feeRate) / 1e5
+	options := &btcjson.FundRawTransactionOpts{
+		FeeRate:    &feeRateOption,
+		ChangeType: &btcjson.ChangeTypeBech32,
+	}
+	if subtract {
+		options.SubtractFeeFromOutputs = []int{0}
+	}
+	if !wc.segwit {
+		options.ChangeType = &btcjson.ChangeTypeLegacy
+	}
+
+	res := &btcjson.FundRawTransactionResult{}
+	err = wc.call(methodFundRawTransaction, anylist{hex.EncodeToString(txBytes), options, false}, &res)
```

Comment:
Bitcoin Cash and Dogecoin don't take an `isWitness` argument or take a  `change_type` in their `options`. Zcash doesn't take `options` at all.

Reply:
> Zcash doesn't take options at all.

Since `Zcash` does not support `options`, there is no means to specify whether the `tx` is a withdrawal or send. Any suggestion on this? Could fee estimate for sweep transactions be affected?

Reply:
I think a question directed at the zcashd devs is why `sendtoaddress` has a subtract option but `fundrawtransaction` does not.

That said, I don't think we should waste any time on a limitation of zcashd in an application where it is just an estimate.  IMO, until they can update the RPCs to match, the (relatively rare) zcash full node dex user can get an inaccurate fee estimate, which won't make much difference in most cases.

**As long as the UI still lets them easily do an actual sweep** without filling in incorrect values, I don't really think we should bend over backwards over this.

Reply:
> I think a question directed at the zcashd devs is why sendtoaddress has a subtract option but fundrawtransaction does not.

I was thinking of this when I read their docs. It's really surprising.